src/main/java/com/hubspot/jinjava/interpret/JinjavaInterpreter.java

```diff
@@ -282,17 +283,22 @@ private void resolveBlockStubs(OutputList output, Stack<String> blockNames) {
     for (BlockPlaceholderOutputNode blockPlaceholder : output.getBlocks()) {
 
       if (!blockNames.contains(blockPlaceholder.getBlockName())) {
-        Collection<List<? extends Node>> blockChain = blocks.get(blockPlaceholder.getBlockName());
-        List<? extends Node> block = Iterables.getFirst(blockChain, null);
+        Collection<BlockInfo> blockChain = blocks.get(blockPlaceholder.getBlockName());
+        BlockInfo block = Iterables.getFirst(blockChain, null);
 
-        if (block != null) {
-          List<? extends Node> superBlock = Iterables.get(blockChain, 1, null);
+        if (block != null && block.getNodes() != null) {
+          List<? extends Node> superBlock = Optional.ofNullable(Iterables.get(blockChain, 1, null))
+              .map(BlockInfo::getNodes).orElse(null);
           context.setSuperBlock(superBlock);
 
           OutputList blockValueBuilder = new OutputList(config.getMaxOutputSize());
 
-          for (Node child : block) {
+          for (Node child : block.getNodes()) {
+            block.getParentPath().ifPresent(path -> getContext().getCurrentPathStack().push(path, lineNumber, position));
+
             blockValueBuilder.addNode(child.render(this));
+
+            block.getParentPath().ifPresent(path -> getContext().getCurrentPathStack().pop());
```

Comment:
nit: path is unused, can be replaced with `()`

Reply:
I'm getting a compile error if do that: `Cannot infer functional interface type`
I think it's because `ifPresent` takes a `Consumer`.